### PR TITLE
KETTLE-57: Fix and test case for failure to properly mark request during action of asynchronous middleware

### PR DESCRIPTION
**Pull request by amb26**

### CODE REVIEW

File: docs/Middleware.md [hunk 1]
```diff
@@ -120,6 +120,13 @@ fluid.defaults("kettle.middleware.json", {
 Consult the Infusion documentation on the [compact format for expanders](http://docs.fluidproject.org/infusion/development/ExpansionOfComponentOptions.html#compact-format-for-expanders) if you
 are unfamiliar with this syntax for designating elements in component options which arise from function calls.
 
+If your middleware may act asynchronously by performing some raw I/O, you must use the grade `kettle.plainAsyncMiddleware`
+instead. This is to ensure that the Kettle [request component](RequestHandlersAndApps.md#request-components)
+is unmarked during the period that the system is not acting on behalf of the currently incoming request. If the code
+for the middleware is under your control, it is recommended that wherever possible you use [dataSources][DataSources.md]
+for I/O since their callbacks automatically perform the necessary
+[request marking](DataSources.md#callback-wrapping-in-dataSources).
```
Comment (simonbates) [inline docs/Middleware.md]: I think that the "dataSources" in the link id should have a lower case "s". I think it should be: DataSources.md#callback-wrapping-in-datasources

File: tests/StaticTests.js [hunk 2]
```diff
@@ -61,6 +68,8 @@ kettle.tests["static"].testDefs = [{
     },
     sequence: [{
         func: "{packageRequest}.send"
+    }, { // Send 2nd request back-to-back to test KETTLE-57
+        func: "{packageRequest2}.send"
```
Comment (simonbates) [inline tests/StaticTests.js]: This test verifies that the request marker is cleared but not that it is set again at the middleware callback ("kettle.markActiveRequest(request)") -- is there any easy/convenient way to verify that the request is marked again upon return from async?

**Comment by amb26** [comment]
Thanks for these great comments, @simonbates - should be all addressed, ready for another round of review!

File: tests/StaticTests.js (hunk 2 above)
Comment (amb26) [inline tests/StaticTests.js]: Good observation - I've added a check in an instrumented version of the "notFoundHandler" bound to the request

**Comment by simonbates** [comment]
Merged at be1f3b2325332e06e9f4591e821e59d59ff153ec

**Pull request merged**